apps/nextjs/src/stores/resourcesStore/actionFunctions/handleSubmitLessonPlan.ts

```diff
@@ -87,6 +93,32 @@ const updateStoreWithLessonPlan = (
   log.info("Lesson plan updated successfully");
 };
 
+/**
+ * Updates the material session with the lesson ID
+ */
+const updateMaterialSessionWithLessonId = async (
+  resourceId: string | null,
+  lessonId: string,
+  updateSessionMutateAsync?: UseMutateAsyncFunction<
+    { success: boolean },
+    Error,
+    { resourceId: string; lessonId: string }
+  >,
+) => {
+  if (resourceId && updateSessionMutateAsync) {
```

Comment:
Is this a valid case? If so wouldn't we want to handle it or log?

If it's just for typescript then we can use an invariant. That way if it's being called incorrectly we will know befofe it gets to production